crates/fluvio-test/src/tests/election/mod.rs

```diff
@@ -52,137 +52,141 @@ impl TestOption for ElectionTestOption {
 }
 
 #[fluvio_test(topic = "test")]
-pub async fn election(
-    mut test_driver: Arc<FluvioTestDriver>,
-    mut test_case: TestCase,
-) -> TestResult {
+pub fn election(mut test_driver: Arc<FluvioTestDriver>, mut test_case: TestCase) -> TestResult {
     println!("Starting election test");
 
-    // first a create simple message
-    let topic_name = test_case.environment.topic_name();
-    let producer = test_driver.create_producer(&topic_name).await;
-
-    producer
-        .send(RecordKey::NULL, "msg1")
-        .await
-        .expect("sending");
-
-    // this is hack now, because we don't have ack
-    sleep(Duration::from_secs(ACK_WAIT)).await;
-
-    let admin = test_driver.client().admin().await;
-
-    let partitions = admin
-        .list::<PartitionSpec, _>(vec![])
-        .await
-        .expect("partitions");
+    run_block_on(async {
```

Comment:
Don't see why we couldn't keep existing signature.  Instead of requiring `run_block_on`, inject that from `fluvio_test` procedure macro so it will be pretty much same as before.  In fact: `test_driver
             .connect().await` can be part of generated code

Reply:
Both the smoke and longevity tests (and any future test where we're going to spawn processes) need to defer their async code until after fork(). I found that we can't async any earlier, just based on my small experimenting: https://github.com/infinyon/fluvio/issues/627#issuecomment-931547608

So that's one reason we wouldn't want to `test_driver.connect().await` in the generated code.